lib/src/rules/library_private_types_in_public_api.dart

```diff
@@ -80,6 +80,14 @@ class Validator extends SimpleAstVisitor<void> {
     node.members.accept(this);
   }
 
+  @override
+  void visitExtensionTypeDeclaration(ExtensionTypeDeclaration node) {
+    if (Identifier.isPrivateName(node.name.lexeme)) return;
+    node.typeParameters?.accept(this);
+    node.representation.fieldType.accept(this);
```

Comment:
I think the representation type only needs to be checked in cases where the name of the representation field is public. That is, I don't think there should be a lint for
```
extension type E(_C _c) {}
class _C {}
```
because the extension field isn't part of the public API in that case.

Reply:
Good catch!